### PR TITLE
[deckhouse-controller] execute opertator.Start() after preflight checks

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
This pr reorders the start-up sequence, making operator.Start() run after controllers' preflight checks.

## Why do we need it, and what problem does it solve?
Currently, operator.Start() is executed before controllers' preflight checks, and we may end up having modules in inconsistent state.

## Why do we need it in the patch release (if we do)?
We need to fix the bug when some modules have different states between file system and the module manager.

## What is the expected result?
The addon-operator starts only after all controllers' preflight checks have been successfully run.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: fix
summary: Start addon-operator after controllers' preflight checks.
impact_level: default
```

### CODE REVIEW

File: deckhouse-controller/cmd/deckhouse-controller/start.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func run(ctx context.Context, operator *addon_operator.AddonOperator) error {
 		return err
 	}
 
-	dController, err := controller.NewDeckhouseController(ctx, operator.KubeClient().RestConfig(), operator.ModuleManager, operator.MetricStorage)
+	dController, err := controller.NewDeckhouseController(ctx, operator.KubeClient().RestConfig(), operator.ModuleManager, operator.MetricStorage, operator.Start)
```
Comment (yalosev) [inline deckhouse-controller/cmd/deckhouse-controller/start.go:180]:
```suggestion
	dController, err := controller.NewDeckhouseController(ctx, operator.KubeClient().RestConfig(), operator.ModuleManager, operator.MetricStorage, operator)
```

File: deckhouse-controller/pkg/controller/controller.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ type DeckhouseController struct {
 	embeddedDeckhousePolicy *v1alpha1.ModuleUpdatePolicySpec
 }
 
-func NewDeckhouseController(ctx context.Context, config *rest.Config, mm *module_manager.ModuleManager, metricStorage *metric_storage.MetricStorage) (*DeckhouseController, error) {
+func NewDeckhouseController(ctx context.Context, config *rest.Config, mm *module_manager.ModuleManager, metricStorage *metric_storage.MetricStorage, runnables ...manager.RunnableFunc) (*DeckhouseController, error) {
```
Comment (yalosev) [inline deckhouse-controller/pkg/controller/controller.go:89]:
```suggestion
func NewDeckhouseController(ctx context.Context, config *rest.Config, mm *module_manager.ModuleManager, metricStorage *metric_storage.MetricStorage, runnables ...manager.Runnable) (*DeckhouseController, error) {
```

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 9253591256 9253591256-1-con-1-25 /sys/deckhouse-oss/install:pr8485 centos@3.75.7.55

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/aws'...

**Comment by miklezzzz** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 9257091059 9257091059-1-con-1-25 /sys/deckhouse-oss/install:pr8485 centos@18.197.99.109

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/aws'...

**Comment by miklezzzz** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 9258051805 9258051805-1-con-1-25 /sys/deckhouse-oss/install:pr8485 centos@18.184.45.191

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit 3bce25f0cc1994eef6341aba37b190c2735f82a1 to the branch [release-1.60](https://github.com/deckhouse/deckhouse/tree/release-1.60) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/9265702440) for details.

**Pull request merged**